python-avd/pyavd/_eos_designs/eos_designs_facts/uplinks.py

```diff
@@ -299,7 +299,11 @@ def _get_l2_uplink(
         if uplink_native_vlan := get(self.shared_utils.switch_data_combined, "uplink_native_vlan"):
             uplink["native_vlan"] = uplink_native_vlan
 
-        if self._short_esi is not None:
+        if self.shared_utils.mlag_role == "secondary":
+            # On the MLAG Secondary use short-esi from MLAG primary
+            if (peer_short_esi := self.shared_utils.mlag_peer_facts._short_esi) is not None:
+                uplink["peer_short_esi"] = peer_short_esi
+        elif self._short_esi is not None:
```

Comment:
I don't think this should happen here but rather here: https://github.com/aristanetworks/avd/blob/775a0de8ac676c15824042732128c774945d2279/python-avd/pyavd/_eos_designs/eos_designs_facts/short_esi.py#L25 otherwise we are computing a wrong fact right?

Reply:
Following @ClausHolbechArista from the issue:


> So we need to ensure that we use the node ID of the first MLAG switch for short_esi: auto.

Reply:
as discussed over chat with @Vibhu-gslab let's wait for @ClausHolbechArista return to discuss this further

Reply:
`shared_utils._short_esi` is only used here and above in some description decision. By moving the condition to the _short_esi method, we address a very unlikely corner-case leading to wrong description, so let's do that.

Reply:
Moved the logic!